index.js

```diff
@@ -708,27 +721,52 @@ Parser.prototype = {
     return {type: 'MixinBlock', line: tok.line, filename: this.filename};
   },
 
+  parseYield: function() {
+    var tok = this.expect('yield');
+    return {type: 'YieldBlock', line: tok.line, filename: this.filename};
+  },
+
   /**
    * include block?
    */
 
   parseInclude: function(){
     var tok = this.expect('include');
-
-    return {
+    var node = {
       type: 'Include',
       file: {
         type: 'FileReference',
-        path: tok.val.trim(),
         line: tok.line,
         filename: this.filename
       },
-      filter: tok.filter,
-      attrs: tok.attrs ? tok.attrs.attrs : [],
-      block: 'indent' == this.peek().type ? this.block() : this.emptyBlock(tok.line),
       line: tok.line,
       filename: this.filename
     };
+    var filters = [];
+    while (this.peek().type === 'filter') {
+      filters.push(this.parseIncludeFilter());
+    }
+    var path = this.expect('path');
+
+    node.file.path = path.val.trim();
+
+    if (/\.jade$/.test(node.file.path)) {
+      node.block = 'indent' == this.peek().type ? this.block() : this.emptyBlock(tok.line);
+      if (filters.length) {
+        // TODO: make this a warning
+        // this.error('Jade inclusion cannot be filtered; filters ignored', 'JADE_INCLUDE_FILTER', path);
+      }
```

Comment:
jade can be filtered.  The condition here should be `/\.jade$/.test(node.file.path) && !filters.length`.  This allows you to do things like:

```jade
h1 Example of a jade file
include:syntax-highlight ./my-jade-file.jade
```